lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -182,14 +187,44 @@ void CDataFrameTrainBoostedTreeClassifierRunner::writeOneRow(
                                 writer.Key(classValues[j]);
                                 writer.Double(shap[i](j));
                             }
-                            writer.Key(CDataFrameTrainBoostedTreeRunner::IMPORTANCE_FIELD_NAME);
+                            writer.Key(IMPORTANCE_FIELD_NAME);
                             writer.Double(shap[i].lpNorm<1>());
                         }
                         writer.EndObject();
                     }
                 }
                 writer.EndArray();
+
+                for (std::size_t i = 0; i < shap.size(); ++i) {
+                    if (shap[i].lpNorm<1>() != 0) {
+                        if (totalShapValues.find(i) != totalShapValues.end()) {
+                            totalShapValues[i] += shap[i].cwiseAbs();
+                        } else {
+                            totalShapValues[i] = shap[i].cwiseAbs();
+                        }
```

Comment:
The way to do this avoiding multiple map lookups is:
```suggestion
                        *totalShapValues.emplace(i, TVector::Zero(shap[i].size())).first += shap[i].cwiseAbs();
```

Reply:
Nice trick!

Reply:
One thing though is `TVector::Zero(shap[i].size())` is allocating; so even better would be create this once upfront outside of the lambda then reference it here when adding to the container. Then you cutout all the extra allocations.